MoreLinq/ListLike.cs

```diff
@@ -35,6 +36,17 @@ static class ListLike
         public static IListLike<T> AsListLike<T>(this IList<T> list) => new List<T>(list);
         public static IListLike<T> AsListLike<T>(this IReadOnlyList<T> list) => new ReadOnlyList<T>(list);
 
+        public static IListLike<T> ToListLike<T>(this IEnumerable<T> source)
```

Comment:
`source` argument should be validated to be non-null.

To avoid too much mixing of styles, let's convert this to a simple expression, [like you did also in PR #499](https://github.com/morelinq/MoreLINQ/pull/499/files#diff-d5ae966de6f42beabe6e1199cd4c4bf3R60):

```c#
public static IListLike<T> ToListLike<T>(this IEnumerable<T> source)
    => source is null ? throw new ArgumentNullException(nameof(source))
     : source is IList<T> list ? new List<T>(list)
     : source is IReadOnlyList<T> readOnlyList ? (IListLike<T>) new ReadOnlyList<T>(readOnlyList)
     : new List<T>(source.ToList());
```

Reply:
update done. the idea is after PR #499 be merged refactore this one to 

```c#
public static IListLike<T> ToListLike<T>(this IEnumerable<T> source)
    => source is null ? throw new ArgumentNullException(nameof(source))
          : source.TryAsListLike() is IListLike<T> listLike
               ? listLike
               : new List<T>(source.ToList());
```